File: src/gestorAplicación/clasesPrincipales/Reserva.java
package gestorAplicación.clasesPrincipales;
import java.io.Serializable;
import java.util.ArrayList;
import gestorAplicación.clasesHeredadas.Cliente;
import gestorAplicación.clasesHeredadas.Trabajador;

public class Reserva implements Serializable {
    private int hora;
	private String IdR;
    private Cliente cliente;
    private Mesa mesa;
	public static ArrayList<Reserva> reservasHechas= new ArrayList<Reserva>();

	public Reserva(int hora, Cliente cliente, Mesa mesa) {
		this.hora = hora;
		this.cliente = cliente;
		this.mesa = mesa;
		this.IdR= hora+"-"+mesa.getId();
		reservasHechas.add(this);
		Trabajador.addMesasElegir(this);
	}
	
	//Métodos get
	public int getHora() {
		return hora;
	}
	public Cliente getCliente() {
		return cliente;
	}
	public Mesa getMesa() {
		return mesa;
	}
	public String getIdR(){
		return IdR;
	}
	public static ArrayList<Reserva> getReservasHechas(){
		return reservasHechas;
	}
	
	//Métodos set
	public void setHora(int hora) {
		this.hora = hora;
	}
	public void setCliente(Cliente cliente) {
		this.cliente = cliente;
	}
	public void setMesa(Mesa mesa) {
		this.mesa = mesa;
	}
	public String toString(){
		return "Cliente: " + this.cliente + ", mesa: " + this.mesa + ", hora: " + this.hora + ", reservaID: " + this.IdR;
	}


	//Recibe una lista de mesas y revisa si estan disponibles en cierto horario
	public static ArrayList<Mesa> validarHorarioDisponible(ArrayList<Mesa> mesas, int hora) {
		ArrayList<Mesa> mesasReservadas = new ArrayList<>();
		for (Mesa m : mesas) {
			for (Reserva r : Reserva.reservasHechas) {
				if (r.getIdR().trim().equals(hora+"-"+m.getId())) {
					mesasReservadas.add(m);
				}
			}
		}
		mesas.removeAll(mesasReservadas);
		return mesas;
	}
		
	public static String devolverSede(int sede){
		String Isede = "";
		if (Sedes.B.ordinal() == sede) {
        	Isede = "B";
        } else if (Sedes.E.ordinal() == sede) {
        	Isede = "E";
        } else if (Sedes.SJ.ordinal() == sede) {
        	Isede = "SJ";
        }
		return Isede;
	}
	
	public void cambiarParametros(int hora, Mesa mesa){
		this.hora = hora;
		this.mesa = mesa;
		this.IdR = hora+"-"+mesa.getId();
	}

	public static Reserva buscarReserva(String id){
		for (Reserva r : reservasHechas){
			if (r.getIdR().trim().equals(id)){
				return r;
			}
		}
		return null;
	}
}

File: Python/src/gestorAplicación/clasesPrincipales/Reserva.py
from gestorAplicación.clasesHeredadas import Trabajador,Cliente
import gestorAplicación.clasesPrincipales

class Reserva:

    reservasHechas = []

    #Constructor
    def __init__(self,hora,cliente,mesa):
        self._hora = hora
        self._cliente = cliente
        self._mesa = mesa
        self._IdR = hora+"-"+mesa.getId()
        Reserva.reservasHechas.append(self)
        Trabajador.addMesasElegir(self)


    #Getters
    @classmethod
    def getReservasHechas():
        return Reserva.reservasHechas
    
    def getHora(self):
        return self._hora
    
    def getCliente(self):
        return self._cliente
    
    def getMesa(self):
        return self._mesa
    
    def getIdR(self):
        return self._IdR
    
    #Setters
    def setCliente(self,cliente):
        self.cliente = cliente

    def setMesa(self, mesa):
        self.mesa = mesa

    def setHora(self, hora):
        self.hora = hora
    
    def __Str__(self):
        return "Cliente: " + self._cliente + ", mesa: " + self._mesa + ", hora: " + self._hora + ", reservaID: " + self._IdR
    
    #Recibe una lista de mesas y revisa si estan disponibles en cierto horario
    @classmethod
    def validarHorarioDisponible(mesas,hora):
        mesasReservadas = []
        for m in mesas:
            for r in Reserva.reservasHechas:
                if r.getIdR().trim() == (hora+"-"+m.getId()):
                    mesasReservadas.append(m)
        for i in mesasReservadas:
            mesas.pop(i)
        return mesas

    
    def cambiarParametros(self, hora, mesa):
        self.hora = hora
        self.mesa = mesa
        self.IdR = hora+"-"+mesa.getId()

    @classmethod
    def buscarReserva(id):
        for r in Reserva.reservasHechas:
            if (r.getIdR().trim() == id):
                return r
        return None